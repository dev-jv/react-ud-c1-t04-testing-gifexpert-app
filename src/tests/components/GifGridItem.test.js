import React from 'react';
import {GifGridItem} from "../../components/GifGridItem";
import { shallow } from 'enzyme';
import '../../setupTests';

// eslint-disable-next-line no-unused-vars
import colors from 'colors';

// eslint-disable-next-line jest/valid-title
describe('Testing in <GifGridItem />'.bgBlue.white, () => {

    const title = 'Any title';
    const url = 'https://localhost/some.jpg';
    let wrapper = shallow(<GifGridItem title={ title } url={ url }/>);

   test('1: Should show <GifGridItem /> correctly',() => {
       expect(wrapper).toMatchSnapshot();
   });

   test('2: Should have a paragraph with the title', () => {
       const p = wrapper.find('p');
       console.log(p.text());
       console.log(p.text().trim());
       expect(p.text().trim()).toBe(title);
   });

    test('3: Should have an image like in the url, alt in props', () => {
        const img = wrapper.find('img');
        // console.log(img.html());
        console.log(img.props().src);
        // expect(img.props().src).toBe(url);
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });

    test('4: Should have an animate__fadeIn', () => {
        const div = wrapper.find('div');
        const className = div.prop('className');
        expect(className.includes('animate__fadeIn')).toBe(true);
    });

});
