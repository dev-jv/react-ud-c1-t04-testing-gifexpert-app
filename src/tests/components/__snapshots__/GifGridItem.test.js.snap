// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[37m[44mTesting in <GifGridItem />[49m[39m 1: Should show <GifGridItem /> correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Any title"
    src="https://localhost/some.jpg"
  />
  <p>
    Any title
  </p>
</div>
`;
